### ISSUE TITLE
Need a public method for migrating database in tests

### ISSUE THREAD

**Issue by schneems** [rails/rails#36544]
For gems that use database migrations, many of them were previously using this code in a `test_helper.rb` file:

```
ActiveRecord::Migrator.migrate File.expand_path("../dummy/db/migrate/", __FILE__)
```

This method was removed and now people are using:

```
 ActiveRecord::MigrationContext.new('spec/dummy/db/migrate').migrate
```

For example https://github.com/influitive/apartment/pull/523/files, however this is also not a public interface and has changed again in 6.1 https://github.com/rails/rails/commit/7cc27d749c3563e6b278ad01d233cb92ea3b7935 which results in more errors

Can we make MigrationContext a public interface or expose some other "blessed" way to explicitly migrate in tests?

**Comment by eileencodes** [comment]
The old method was private too right? I was pretty sure we didn't change the API when I added that.

**Comment by schneems** [comment]
I think you're correct, it's just one of those things people were using from some stack overflow article or tutorial for a long time and no one really stopped to check if it was an official interface or it just happened to work. I honestly don't eve know where I got the `ActiveRecord::Migrator.migrate File.expand_path("../dummy/db/migrate/", __FILE__)` snippet from, but I've got it in a few projects. 

Currently I'm fighting with it in https://github.com/schneems/derailed_benchmarks/blob/088a53e181e6f264b96e21c2a4c75487b586d4e4/test/test_helper.rb#L29-L36

**Comment by eileencodes** [comment]
Do you want to add a public API or wait for me to do it? It's on my list but lower down compared to some of the public 🐛s I created 😆

**Comment by schneems** [comment]
I'm hunting to see if there's a public way that currently works right now, so I don't think it will be a huge blocker if I can work around it in the short term.

I'll ping you before I start any work on it so we don't overlap. Also if you want any eyes here or want to remote pair, I can put something on the schedule.

**Comment by thomasdziedzic** [comment]
I use `ActiveRecord::Migration.maintain_test_schema!`
It also seems like there is a config option for it `config.active_record.maintain_test_schema` mentioned in https://guides.rubyonrails.org/configuring.html

**Comment by schneems** [comment]
@thomasdziedzic what does calling that do exactly in the context of a engine/library? I tried calling it in derailed_benchmarks and it seemed to do nothing. I ended up using this incantation cribbed from devise https://github.com/schneems/derailed_benchmarks/blob/663be3db7aa9c10efb520c680b54fb33bc954813/test/test_helper.rb#L33-L41

I think "all" we need to do to resolve this issue is make `ActiveRecord::MigrationContext#migrate` public, add some method docs, and document it in the guides engines section.
